taichi/program/program.cpp

```diff
@@ -564,6 +564,22 @@ Ndarray *Program::create_ndarray(const DataType type,
   return ndarrays_.back().get();
 }
 
+Texture *Program::create_texture(const DataType type,
+                                 int num_channels,
+                                 const std::vector<int> &shape) {
+  if (shape.size() == 1) {
+    textures_.emplace_back(this, type, num_channels, shape[0], 1, 1);
+  } else if (shape.size() == 2) {
+    textures_.emplace_back(this, type, num_channels, shape[0], shape[1], 1);
+  } else if (shape.size() == 3) {
+    textures_.emplace_back(this, type, num_channels, shape[0], shape[1],
+                           shape[2]);
+  } else {
+    TI_ERROR("Texture shape invalid");
+  }
+  return &textures_.back();
```

Comment:
You don't have pointer stability here, consider using `std::vector<std::unique_ptr<Texture>>`

Reply:
ooo right I forgot about that, was wondering why the ndarrays uses a wrapped pointer